spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/dsl/CheckPointedTaskDefinitionException.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.core.dsl;
+
+import java.util.List;
+
+/**
+ * An extension of TaskDefinitionException that remembers the point up
+ * to which parsing went OK (signaled by a '*' in the dumped message).
+ *
+ * @author Eric Bottard
+ */
+@SuppressWarnings("serial")
+public class CheckPointedTaskDefinitionException extends TaskDefinitionException {
```

Comment:
if this is a direct copy of CheckPointedStreamDefinitionException, I'd prefer we make the Exception hierarchy more general and reuse the same